deepchecks/nlp/checks/multivariate_embeddings_drift_utils.py

```diff
@@ -0,0 +1,120 @@
+# ----------------------------------------------------------------------------
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module containing common EmbeddingsDrift Check (domain classifier drift) utils."""
+
+from typing import Tuple
+
+import pandas as pd
+from sklearn.decomposition import PCA
+from sklearn.ensemble import GradientBoostingClassifier
+from sklearn.metrics import roc_auc_score
+from sklearn.model_selection import train_test_split
+from umap import UMAP
+
+from deepchecks.core.check_utils.multivariate_drift_utils import auc_to_drift_score, build_drift_plot
+from deepchecks.nlp import TextData
+from deepchecks.nlp.utils.nlp_plot import two_datasets_scatter_plot
+from deepchecks.utils.dataframes import floatify_dataframe
+from deepchecks.utils.plot import DEFAULT_DATASET_NAMES
+
+SAMPLES_FOR_REDUCTION_FIT = 1000
+
+
+def run_multivariable_drift_for_embeddings(train_dataset: TextData, test_dataset: TextData,
+                                           sample_size: int, random_state: int, test_size: float,
+                                           min_meaningful_drift_score: float, num_samples_in_display: int,
+                                           dimension_reduction_method: str, with_display: bool,
+                                           dataset_names: Tuple[str] = DEFAULT_DATASET_NAMES):
+    """Calculate multivariable drift on embeddings."""
+    # sample train and test datasets equally
+    train_sample = train_dataset.sample(sample_size, random_state=random_state)
+    test_sample = test_dataset.sample(sample_size, random_state=random_state)
+
+    train_sample_df = train_sample.embeddings
+    test_sample_df = test_sample.embeddings
+
+    # create new dataset, with label denoting whether sample belongs to test dataset
+    domain_class_df = pd.concat([train_sample_df, test_sample_df]).reset_index(drop=True)
+    domain_class_labels = pd.Series([0] * len(train_sample_df) + [1] * len(test_sample_df))
+
+    # reduce dimensionality of embeddings if needed.
+    # skips if not required ('none') or if number of features is small enough (< 30) in 'auto' mode.
+    if not (dimension_reduction_method == 'none' or (
+            dimension_reduction_method == 'auto' and domain_class_df.shape[1] < 30)):
+
+        if (dimension_reduction_method == 'auto' and with_display) or dimension_reduction_method == 'umap':
```

Comment:
Would suggest for clarity to put all these conditions into variables before entering the if code.